File: core 2/src/main/java/com/dev/core/service/ProjectFileService.java
package com.dev.core.service;


import com.dev.core.model.ProjectFileDTO;
import org.springframework.web.multipart.MultipartFile;
import java.util.List;

public interface ProjectFileService {

    /**
     * Upload and attach a file to a project.
     */
    ProjectFileDTO uploadFile(Long projectId, MultipartFile file, String description, String visibility);

    /**
     * List all files belonging to a project.
     */
    List<ProjectFileDTO> getFilesByProject(Long projectId);

    /**
     * Get a specific file record by ID.
     */
    ProjectFileDTO getFileById(Long id);

    /**
     * Delete a file and remove metadata (soft delete).
     */
    void deleteFile(Long id);

    /**
     * Generate a secure download URL or local path for serving the file.
     */
    String getDownloadUrl(Long fileId);
}

File: core-platform/apps/core-webapp/src/services/projectFile.service.ts
import axiosInstance from '../axiosInstance';

const PROJECT_FILE_API_BASE = '/projects/files';

export interface ProjectFileDTO {
    id: number;
    projectId: number;
    fileName: string;
    originalFileName: string;
    fileSize: number;
    fileType: string;
    filePath: string;
    description?: string;
    visibility: 'INTERNAL' | 'EXTERNAL';
    uploadedBy: number;
    uploadedByName?: string;
    uploadedAt: string;
}

export const projectFileService = {
    // Upload file to project
    uploadFile: async (
        projectId: number,
        file: File,
        description?: string,
        visibility: 'INTERNAL' | 'EXTERNAL' = 'INTERNAL'
    ): Promise<ProjectFileDTO> => {
        const formData = new FormData();
        formData.append('file', file);
        if (description) {
            formData.append('description', description);
        }
        formData.append('visibility', visibility);

        const response = await axiosInstance.post(
            `${PROJECT_FILE_API_BASE}/${projectId}/upload`,
            formData,
            {
                headers: {
                    'Content-Type': 'multipart/form-data',
                },
            }
        );
        return response.data.data;
    },

    // Get all files for a project
    getFilesByProject: async (projectId: number): Promise<ProjectFileDTO[]> => {
        const response = await axiosInstance.get(`${PROJECT_FILE_API_BASE}/project/${projectId}`);
        return response.data.data;
    },

    // Get file by ID
    getFileById: async (id: number): Promise<ProjectFileDTO> => {
        const response = await axiosInstance.get(`${PROJECT_FILE_API_BASE}/${id}`);
        return response.data.data;
    },

    // Delete file
    deleteFile: async (id: number): Promise<void> => {
        await axiosInstance.delete(`${PROJECT_FILE_API_BASE}/${id}`);
    },

    // Get download URL
    getDownloadUrl: async (id: number): Promise<string> => {
        const response = await axiosInstance.get(`${PROJECT_FILE_API_BASE}/${id}/download-url`);
        return response.data.data;
    }
};

export default projectFileService;
